lib/nve/nve-main-page.dart

```diff
@@ -119,30 +125,77 @@ class FirstScreen extends State<MyStatefulWidget> {
   _onItemTapped(int index) {
     setState(() {
       switch (index) {
-        case 0 :
+        case BLUE_PAGE_INDEX:
           _alignment = Alignment.bottomLeft;
+          _image = Image.asset(
+            "assets/images/mansion.png",
+            width: 50,
+          );
           break;
-        case 1 :
+        case RED_PAGE_INDEX:
           _alignment = Alignment.bottomCenter;
+          _image = Image.asset(
+            "assets/images/map.png",
+            width: 50,
+          );
           break;
-        case 2 :
+        case GREEN_PAGE_INDEX:
           _alignment = Alignment.bottomRight;
+          _image = Image.asset(
+            "assets/images/gear.png",
+            width: 50,
+          );
           break;
       }
       _selectedIndex = index;
+      animatePageChange();
     });
   }
 
+  animatePageChange() async {
+    await _pageViewController.animateToPage(
+        _selectedIndex,
+        duration: Duration(milliseconds: 500),
+        curve: Curves.decelerate,
+      );
+  }
+
   @override
   Widget build(BuildContext ctxt) {
     return new Scaffold(
-      body: _children[_selectedIndex],
+      body: Stack(children: <Widget>[
+        PageView(
```

Comment:
may be you should surround the PageView with some Padding or Positioned to avoid page content hiddent by header.

=> When you focus the text field in the green page, the text field goes under the header and is no longer visible.